daffodil-core/src/main/scala/org/apache/daffodil/dsom/SchemaSet.scala

```diff
@@ -472,7 +472,11 @@ final class SchemaSet(
       }
 
       val newNS = matchingDVs.head.namespace
-      val newBinding = Binding(v.varQName.local, Some(newNS), v.varValue)
+      val newBinding = try {
+        Binding(v.varQName.local, Some(newNS), v.varValue)
+      } catch {
+        case e: BindingException => this.SDE("Exception when processing external variable %s: %s", v.varQName, e.getMessage)
```

Comment:
I think the error message ends up a little verbose. The BindingException has a string like:
```Exception when processing external variable binidng file: %s```
So combine that with the SDE and the message would look like:
```Exception when processing external variable %s: Exception when processing external variable binidng file: %s"```
So the message would end up duplicated, I think.